File: view_based_projects/03-2-OpenCloseActsApp/app/src/main/java/com/example/opencloseactsapp/MainActivity.java
package com.example.opencloseactsapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

import com.example.opencloseactsapp.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.oneButton.setOnClickListener(view -> {
            Intent i = new Intent(MainActivity.this, OneActivity.class);
            i.putExtra("origin", getResources().getString(R.string.main_title));
            startActivity(i);
        });

        binding.twoButton.setOnClickListener(view -> {
            Intent i = new Intent(MainActivity.this, TwoActivity.class);
            startActivity(i);
        });

        binding.exitButton.setOnClickListener(view -> {
            MainActivity.this.finish();
        });
    }
}

File: 03-1-login_kt2/app/src/main/java/com/example/login_kt2/MainActivity.kt
package com.example.login_kt2

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import com.example.login_kt2.ui.theme.Login_kt2Theme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            Login_kt2Theme {
                App(startingRoute = "login")
            }
        }
    }
}